### PR TITLE
Adding ability to read YRBSS files

### PR DESCRIPTION
**Pull request by khwilson**
Hi y'all! Long time no commit :-)

I was trying to read in the various YRBSS data that IPUMS makes available and realized it was in a different format than the other files IPUMS makes available. Since I had to write some code to read it anyway, I figured I'd add it to IPUMS-py.

Hope all is well with you all!

### CODE REVIEW

**Comment by renae-r** [comment]
Thanks so much @khwilson! 

Sorry it took me so long to address this PR! I made some adjustments to make the handling of "non-extractable" IPUMS datasets a little more generalized so that it will work with [IPUMS NYTS](https://www.ipums.org/projects/ipums-nyts) too. (And hopefully any future similar datasets, should they come into existence). 

Does this implementation still meet your needs?

File: src/ipumspy/noextract.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            ],
+            samples_description=[],
+            ipums_citation=j["citation"],
+            ipums_collection=f"{collection}",
+            ipums_conditions="",
+            ipums_doi="",
+        )
+
+
+def download_noextract_data(collection: str, filename: FilenameType):
```
Comment (khwilson) [inline src/ipumspy/noextract.py:74]: Potentially you might want to allow for a default filename?

File: tests/test_noextract.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+import pytest
+import vcr
+
+from ipumspy import noextract
+
+# TODO
+# [] mock download of non-extractable data
```
Comment (khwilson) [inline tests/test_noextract.py:8]: Extraneous (?) TODO

**Comment by khwilson** [comment]
So glad it was extensible! LGTM. I just had a couple very tiny comments.

**Review by renae-r — APPROVED** [review]

**Comment by renae-r** [comment]
Woot!

**Pull request merged**